frontends/web/src/routes/account/add/addaccount.tsx

```diff
@@ -21,41 +21,103 @@ import { Button, ButtonLink, Input, Select } from '../../../components/forms';
 import { Entry } from '../../../components/guide/entry';
 import { Guide } from '../../../components/guide/guide';
 import Header from '../../../components/header/Header';
+import { load } from '../../../decorators/load';
 import { translate, TranslateProps } from '../../../decorators/translate';
 import { apiPost } from '../../../utils/request';
 
+const COIN_AND_ACCOUNT_CODES = {
+    'btc-p2wpkh-p2sh': {
+        name: 'Bitcoin',
+        coinCode: 'btc',
+        scriptType: 'p2wpkh-p2sh',
+    },
+    'btc-p2wpkh': {
+        name: 'Bitcoin: bech32',
+        coinCode: 'btc',
+        scriptType: 'p2wpkh',
+    },
+    'btc-p2pkh': {
+        name: 'Bitcoin Legacy',
+        coinCode: 'btc',
+        scriptType: 'p2pkh',
+    },
+    'ltc-p2wpkh-p2sh': {
+        name: 'Litecoin',
+        coinCode: 'ltc',
+        scriptType: 'p2wpkh-p2sh',
+    },
+    'ltc-p2wpkh': {
+        name: 'Litecoin: bech32',
+        coinCode: 'ltc',
+        scriptType: 'p2wpkh',
+    },
+    // Testnet
+    'tbtc-p2wpkh-p2sh': {
+        name: 'Bitcoin Testnet',
+        coinCode: 'tbtc',
+        scriptType: 'p2wpkh-p2sh',
+    },
+    'tbtc-p2wpkh': {
+        name: 'Bitcoin Testnet: bech32',
+        coinCode: 'tbtc',
+        scriptType: 'p2wpkh',
+    },
+    'tbtc-p2pkh': {
+        name: 'Bitcoin Testnet Legacy',
+        coinCode: 'tbtc',
+        scriptType: 'p2pkh',
+    },
+    'tltc-p2wpkh-p2sh': {
+        name: 'Litecoin Testnet',
+        coinCode: 'tltc',
+        scriptType: 'p2wpkh-p2sh',
+    },
+    'tltc-p2wpkh': {
+        name: 'Litecoin Testnet: bech32',
+        coinCode: 'tltc',
+        scriptType: 'p2wpkh',
+    },
+};
+
+
+
 interface State {
-    coinCode: string;
-    scriptType: string;
+    value: string;
     accountName: string;
     extendedPublicKey: string;
 }
 
-class AddAccount extends Component<TranslateProps, State> {
-    constructor(props: TranslateProps) {
+interface TestingProps {
+    testing?: boolean;
+}
+
+type Props = TestingProps & TranslateProps;
+
+class AddAccount extends Component<Props, State> {
+    constructor(props: Props) {
         super(props);
 
         this.state = {
-            coinCode: 'tbtc',
-            scriptType: 'p2wpkh-p2sh',
+            value: props.testing ? 'tbtc-p2wpkh-p2sh' : 'btc-p2wpkh-p2sh',
             accountName: '',
             extendedPublicKey: '',
         };
     }
 
     private submit = () => {
-        const body = {
-            coinCode: this.state.coinCode,
-            scriptType: this.state.scriptType,
+        const { coinCode, scriptType } = COIN_AND_ACCOUNT_CODES[this.state.value];
```

Comment:
I'm not sure whether `coinCode, scriptType` are typed here. Please verify in IDE by hovering over the names. If not, we could/should improve the typing of `COIN_AND_ACCOUNT_CODES`. But maybe typing `this.state.value` as suggested above already helped.

Reply:
Seems as it only knows that this should be a string